iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -217,15 +215,14 @@ private static void writeExtensions(Network n, NetworkXmlWriterContext context,
 
             Collection<? extends Extension<? extends Identifiable<?>>> extensions = identifiable.getExtensions().stream()
                     .filter(e -> options.withExtension(e.getName()))
+                    .filter(e -> !e.isEmpty())
                     .filter(e -> getExtensionXmlSerializer(options, e.getName()) != null)
                     .collect(Collectors.toList());
             if (!extensions.isEmpty()) {
                 context.getExtensionsWriter().writeStartElement(context.getVersion().getNamespaceURI(), EXTENSION_ELEMENT_NAME);
                 context.getExtensionsWriter().writeAttribute(ID, context.getAnonymizer().anonymizeString(identifiable.getId()));
                 for (Extension<? extends Identifiable<?>> extension : extensions) {
-                    if (options.withExtension(extension.getName())) {
-                        writeExtension(extension, context);
-                    }
```

Comment:
This is a regression?

Reply:
I don't think so, it's always true because of the filter a few lines above

Reply:
Good catch